src/compile/scale.ts

```diff
@@ -65,31 +65,28 @@ export function type(fieldDef: FieldDef, channel: Channel, mark: Mark): string {
         // Also, if we support color ramp, this should be ordinal too.
         return 'time'; // time has order, so use interpolated ordinal color scale.
       }
-      if (contains([ROW, COLUMN, SHAPE], channel)) {
-        return 'ordinal';
-      }
+
       if (fieldDef.scale.type !== undefined) {
         return fieldDef.scale.type;
       }
 
+      // Returns ordinal if
+      // (1) the channel is ROW, COLUMN, or SHAPE, or
+      // (2) uses a BAR or TICK mark
+      // because in that case we should not use a time scale.
+      if (contains([ROW, COLUMN, SHAPE], channel) || contains([BAR, TICK], mark)) {
```

Comment:
I'll test if it makes sense when I arrive at the apartment, gotta go to post office first.